docs/template-development.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Template"
```

Comment:
this document overlaps with https://github.com/asyncapi/generator/blob/master/docs/authoring-templates.md

this one is actually better than the other one, but instead of removing the other and adding new, I suggest we stick to `authoring-templates.md` and just remove what is there and replace what you have here

makes sense?

Reply:
@pratik2315 I think you missed this comment